pyro/infer/importance.py

```diff
@@ -44,3 +44,33 @@ def _traces(self, *args, **kwargs):
                 poutine.replay(self.model, trace=guide_trace)).get_trace(*args, **kwargs)
             log_weight = model_trace.log_prob_sum() - guide_trace.log_prob_sum()
             yield (model_trace, log_weight)
+
+    def get_log_normalizer(self):
+        """
+        Estimator of the normalizing constant of the target distribution.
+        (mean of the unnormalized weights)
+        """
+        log_w = torch.tensor(self.log_weights)
+        log_num_samples = torch.log(torch.tensor(self.num_samples, dtype=torch.float))
+        return torch.logsumexp(log_w - log_num_samples, 0)
+
+    def get_normalized_weights(self, log_scale=False):
+        """
+        Compute the normalized importance weights.
+        """
+        log_w = torch.tensor(self.log_weights)
+        log_w_norm = log_w - torch.logsumexp(log_w, 0)
```

Comment:
i think you might need to use the `logsumexp` in `pyro.distributions.util` for backwards compatibility

Reply:
Sure, will make the change.

Reply:
This is an easy fix. However, I just noticed that pyro.distributions.util just calls torch.logsumexp; are you sure the change is needed?

Reply:
note the `except: ...` for earlier versions, torch doesnt support `logsumexp` which is why there's a hand implementation underneath :)

Reply:
Odd, I do not see the "except:" in my utils.py, only the following: 

```
from __future__ import absolute_import, division, print_function

import numbers
from contextlib import contextmanager

import torch
import torch.distributions as torch_dist
from torch import logsumexp
from torch.distributions.utils import broadcast_all


_VALIDATION_ENABLED = False

log_sum_exp = logsumexp  # DEPRECATED
```

In any case, now logsumexp from this pyro.distributions.util is used.